pkg/kubectl/resource/builder.go

```diff
@@ -155,6 +167,35 @@ func (b *Builder) FilenameParam(enforceNamespace bool, filenameOptions *Filename
 	return b
 }
 
+// Local wraps the builder's clientMapper in a DisabledClientMapperForMapping
+func (b *Builder) Local(mapperFunc ClientMapperFunc) *Builder {
+	if b.isUnstructured {
+		b.errs = append(b.errs, LocalUnstructuredBuilderError)
+		return b
+	}
+
+	b.isLocal = true
+	b.mapper.ClientMapper = DisabledClientForMapping{ClientMapper: ClientMapperFunc(mapperFunc)}
+	return b
+}
+
+// Unstructured updates the builder's ClientMapper, RESTMapper,
+// ObjectTyper, and codec for working with unstructured api objects
+func (b *Builder) Unstructured(mapperFunc ClientMapperFunc, mapper meta.RESTMapper, typer runtime.ObjectTyper) *Builder {
+	if b.isLocal {
```

Comment:
This is not the right place to check this.  Local and unstructured have nothing to do with each other.  Local commands should absolutely be able to function without needing to have compiled in objects.

Reply:
@smarterclayton Thanks, will make a patch to address this